src/main/java/core/org/akaza/openclinica/service/managestudy/StudySubjectServiceImpl.java

```diff
@@ -532,6 +544,44 @@ public void populateUncompletedCRFsWithCRFAndVersions(DataSource ds, ArrayList u
         }
     }
 
+
+
+    public Boolean isSignable(StudyUserRoleBean currentRole, int studySubjectId) {
```

Comment:
I think the role check should happen in the calling method. This method should just check if the subject is signable not include the check for if a user can sign the subject.